dtests/src/test/java/io/snappydata/hydra/cdcConnector/SnappyCDCTest.java

```diff
@@ -689,6 +757,84 @@ public void removeDiskStore() {
     removeDiskStoreFiles(dirPath);
   }
 
+  public static void HydraTask_executeCleanUpFiles() {
+    if (snappyCDCTest == null) {
+      snappyCDCTest = new SnappyCDCTest();
+    }
+    snappyCDCTest.executeCleanUpFiles();
+  }
+
+  public void executeCleanUpFiles() {
+    try {
+      String snappyPath = SnappyCDCPrms.getSnappyFileLoc();
+      File log = new File(".");
+      String dest = log.getCanonicalPath() + File.separator + "clusterRestart.log";
+      File logFile = new File(dest);
+      Map<String,Integer> tableValueMap = new HashedMap();
+      String tableName= "";
+      int expectedCnt =0;
+      String cmd = "rm -rf " + snappyTest.getCurrentDirPath() + "/randomFileForDiskFull_*";
+      ProcessBuilder pb = new ProcessBuilder("/bin/bash", "-c", cmd);
+      snappyTest.executeProcess(pb, logFile);
+      Log.getLogWriter().info("Restarting the cluster after cleanup");
+      stopCluster(snappyPath, logFile);
+      //wait for 5 mins
+      //Thread.sleep(300000);
+      Thread.sleep(5000);
+      startCluster(snappyPath, logFile, false);
+      Log.getLogWriter().info("Verify the total inserts");
+      tableValueMap = (HashedMap)SnappyBB.getBB().getSharedMap().get("TABLE_VALUE_MAP");
+      for (Map.Entry<String,Integer> entry : tableValueMap.entrySet()) {
+        tableName = entry.getKey();
+        expectedCnt = entry.getValue();
+        //int expectedCnt = (Integer) SnappyBB.getBB().getSharedMap().get("TOTAL_NUM_INSERTS");
+        Connection conn = SnappyTest.getLocatorConnection();
+        ResultSet rs = conn.createStatement().executeQuery("SELECT count(*) FROM " + tableName);
+        while (rs.next()) {
+          int actualCnt = rs.getInt(1);
+          if (actualCnt >= expectedCnt) {
```

Comment:
Also, validate for actual row count should not exceed the expected row count + batch size.